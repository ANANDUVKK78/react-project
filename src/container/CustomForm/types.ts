import { FormEventHandler } from "react";

export type Props = {
    formSubmit?: FormEventHandler<HTMLFormElement>
    children: JSX.Element[] | JSX.Element,
    className: string
}
