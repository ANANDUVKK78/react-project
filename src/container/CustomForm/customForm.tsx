import React, { ReactElement } from 'react'
import { Props } from './types'

const CustomForm = (props: Props): ReactElement => {
    return (
        <>
            <form className={props.className} onSubmit={props.formSubmit}>
                {props.children}
            </form>
        </>
    )
}


export default CustomForm
