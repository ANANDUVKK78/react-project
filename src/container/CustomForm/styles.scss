
.form-cntnr{
    min-height:150px;
    border:0.5px solid;
    border-radius: 15px;
    padding:20px
}
