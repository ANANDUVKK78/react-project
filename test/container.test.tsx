import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import CustomForm from 'container';

configure({ adapter: new Adapter() })

describe('Testing Custom form', () => {
    const mockSubmitFunc: jest.Mock = jest.fn()
    it('testing form component', () => {
        const wrapper = shallow(
            <CustomForm formSubmit={mockSubmitFunc} className="sample">
                <button type="submit"></button>
            </CustomForm>
        )
        wrapper.find('form').simulate('submit')
        expect(wrapper.find('form')).toHaveLength(1)
        expect(wrapper.find('button')).toHaveLength(1)
        expect(wrapper.find('form').props().className).toBe('sample')
        expect(mockSubmitFunc.mock.calls.length).toEqual(1)
        const tree = renderer
            .create(
                <CustomForm formSubmit={mockSubmitFunc} className="sample">
                    <button type="submit"></button>
                </CustomForm>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
