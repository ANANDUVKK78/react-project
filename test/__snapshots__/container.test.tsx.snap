// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Custom form testing form component 1`] = `
<form
  className="sample"
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    type="submit"
  />
</form>
`;
